http://sources.redhat.com/ml/libc-alpha/2010-01/msg00038.html

2010-01-17  Samuel Thibault  <samuel.thibault@ens-lyon.org>

	* sysdeps/mach/hurd/bits/libc-lock.h (__rtld_lock_recursive_t): New
	type.
	(__rtld_lock_initialize): New macro.

---
 sysdeps/mach/hurd/bits/libc-lock.h |    3 +++
 1 file changed, 3 insertions(+)

--- a/sysdeps/mach/hurd/bits/libc-lock.h
+++ b/sysdeps/mach/hurd/bits/libc-lock.h
@@ -34,6 +34,7 @@
   void *owner;
   int count;
 } __libc_lock_recursive_t;
+typedef __libc_lock_recursive_t __rtld_lock_recursive_t;
 
 #define __libc_lock_owner_self() ((void *) __hurd_threadvar_location (0))
 
@@ -124,6 +125,8 @@
 
 #define __rtld_lock_init_recursive(NAME) \
   __libc_lock_init_recursive (NAME)
+#define __rtld_lock_initialize(NAME) \
+  (void) ((NAME) = (__rtld_lock_recursive_t) _RTLD_LOCK_RECURSIVE_INITIALIZER)
 #define __rtld_lock_trylock_recursive(NAME) \
   __libc_lock_trylock_recursive (NAME)
 #define __rtld_lock_lock_recursive(NAME) \
